osu.Game/Online/LocalUserStatisticsProvider.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Game.Extensions;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Rulesets;
+using osu.Game.Users;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// A component that is responsible for providing the latest statistics of the logged-in user for the game-wide selected ruleset.
+    /// </summary>
+    public partial class LocalUserStatisticsProvider : Component
+    {
+        /// <summary>
+        /// The statistics of the logged-in user for the game-wide selected ruleset.
+        /// </summary>
+        public IBindable<UserStatistics?> Statistics => statistics;
+
+        private readonly Bindable<UserStatistics?> statistics = new Bindable<UserStatistics?>();
+
+        [Resolved]
+        private IBindable<RulesetInfo> ruleset { get; set; } = null!;
```

Comment:
https://github.com/ppy/osu/pull/27128#discussion_r1486498229 is relevant.

Reply:
I don't really think that comment is relevant. The question here is more of an architectural one no? Like you can still cache the per-ruleset statistics, but instead of exposing the ruleset-specific bindable it would be up to the consumers (`DiscordRichPresence` and `UserRankPanel`).

*That said*, I think I'm okay with it existing as it is. At least until a use case comes up where we want to retrieve this for something that isn't the current global ruleset. Which arguable exists – the single usage you've piped through `GetStatisticsFor`. Is there a reason this flow doesn't do the online retrieval?

Reply:
I will say it does feel weird having this exist alongside `UserStatisticsWatcher` with data being awkwardly ferried between the two. Like they are both basically doing the same thing and the cross-communication is only required to keep the new component in sync?

Reply:
For the sake of making each component have not but a single purpose, LocalUserStatisticsProvider and UserStatisticsWatcher have been split. The first exposes local user statistics to the game codebase, while the second has a very specific purpose tied to score submission which is to listen to submission events and display a change in user statistics while updating the former component with the new statistics.

I can understand that there's still a level of awkwardness between the two components since they both look up statistics online, and I wouldn't mind merging them together, but I would have to hear @bdach's opinion on that since he doesn't like that to be the case.

Reply:
I'm more focused on the fact that you added cross-talk between them, for the sake of saving one request I guess? If we're keeping both, I'd see that being removed.

Reply:
The cross talk must exist for score submission to update the statistics exposed by `LocalUserStatisticsProvider`, the only way to avoid that is merging both components together.

Reply:
My position on this is unchanged: I would like the bindable to be gone.

The reasoning is extremely simple: when faced between two choices, one of which introduces ambient global state, and one of which does not, *always* choose the one that does not introduce the ambient global state unless you absolutely *have to*. It is simpler.

As I see it, `LocalUserStatisticsProvider` should be a dumb lookup facilitator / a _cache_. Same way something like `DifficultyCache` is. It should expose a dictionary for anyone to use in any way they see fit. Keyed by `(userId, rulesetId)`.

As for the interaction with `UserStatisticsWatcher` - `LocalUserStatisticsProvider` should expose a method to _invalidate and refetch_ data for a given cache key, and `UserStatistiscs` should use _that_ to retrieve its update.

Reply:
Sounds good, will apply as proposed.